xmppserver/src/main/java/org/jivesoftware/openfire/spi/ConnectionListener.java

```diff
@@ -322,7 +332,19 @@ public synchronized void restart()
         }
         finally
         {
-            start(); // won't actually start anything if not enabled.
+            // Protect against race conditions when restarting (OF-1956).
+            if ( pauseDurationDuringRestart != null && !pauseDurationDuringRestart.getValue().isNegative() )
+            {
+                try {
+                    Log.debug( "Sleeping {} before restarting...", pauseDurationDuringRestart.getValue() );
+                    Thread.sleep( pauseDurationDuringRestart.getValue().toMillis() );
```

Comment:
I am lukewarm on this approach.  I'd like to see `start()` catch the `BindException` and then only apply the sleep, then try once  more.  Otherwise, one is left wondering how long a duration one needs to tune for this bind exception to go away.  I'd then think the default should be longer than 500 millis.